### ISSUE TITLE
add a property "data source" where you can pass whatever you want

### ISSUE THREAD

**Issue by team-moeller** [team-moeller/better-access-charts#14]
one single data source property where you can pass whatever you want: table name, query name, recordset, sql statement
The correct type is determined by vba

**Comment by juengling** [comment]
You may specify `Public Property Set DataSource(...)` to retrieve a Recordset, or `Public Property Let DataSource(...)` for any String ("object name" or "sql"), but I can't find a method to do both with the same code.

**Comment by team-moeller** [comment]
It is just an idea.
At the moment I have no concrete plans how to solve this problem.
I just wanted to save the idea.
Maybe I re-implement all this in a programming language where overwriting is supported ...

**Comment by juengling** [comment]
Well, if we would add a Sub instead of a property, a parameter with a `variant` datatype can be used, which then handles any distinguishing in the VBA code. I'll try this in my fork, you don't mind?

**Comment by juengling** [comment]
Yes, it's possible. Which source code control tool do you like more: [Ivercy](https://ivercy.com/) or [OASIS](https://dev2dev.de)?

**Comment by KDGundermann** [comment]
and as setting a DataSource without setting DataSourceLabelFieldName and This_DataSourceLabelFieldName all three should go together as Parameters:

`Public Sub setDataSource(DataSource as Variant, DataSourceLabelFieldName as String, This_DataSourceLabelFieldName  as String)`
